### ISSUE TITLE
Legacy points hook extensions not working properly

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/wordpoints#435]
While working on #388, I've discovered that our legacy hook extensions for integrating with the old points hooks do not actually work correctly. The problem is that they are looking for the wrong points log meta keys. They assume that the entity ID will be stored under a key that is like the entity slug, or if the entity is namespaced, like the namespace of the entity slug. So for example it would look for the `post` meta key for the `post\post` or `post\page` entities, and the `user` meta key for the `user` entity. However, this is wrong: the post ID isn't stored under the `post` meta key, but `post_id`. The `register` points hook didn't even store the user ID as metadata, since the ID would have always matched the user ID for the log itself. Likewise for the `periodic` points hook. For the `comment` and `comment_received` hooks, the `comment_id` meta key was used. So I think what we need to do is allow the legacy meta key to query for to be set as reaction metadata, just as we do for the legacy points log type.

**Comment by JDGrimes** [comment]
Should we also switch the new reactor to saving the metadata under the old format? I think not, since it is simpler to be able to just query by entity slug going forward. Really using the old format would also entail special handling for namespaced entities, and I don't think that really is a step forward.

**Comment by JDGrimes** [comment]
However, we might ask whether we should update the old logs on update. That would bring them in line with the new meta key format. Otherwise anything that uses the meta keys is going to need back compat. The only problem though is that we're still maintaining the old hooks for now. We wouldn't be able to actually update this until they were removed entirely, because they rely on the old meta keys in order to work. That is, unless we also updated the old points hook handlers to use the new meta keys. However, that seems like a lot to do right before release, and might break custom code. We still have to worry about back-compat for custom code in regard to the old hooks, even though it obviously won't integrate with the new API automatically, because we aren't importing the old hooks on legacy sites yet. And I guess we really can't do anything here until we do.
